src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8302,7 +8300,95 @@ private ErrorPropertySymbol CreateErrorPropertySymbol(ImmutableArray<PropertySym
             }
         }
 
-        internal bool ReportDelegateInvokeUseSiteDiagnostic(BindingDiagnosticBag diagnostics, TypeSymbol possibleDelegateType,
+#nullable enable
+        internal (NamedTypeSymbol?, MethodSymbol?) GetMethodGroupDelegateType(BoundMethodGroup node, BindingDiagnosticBag diagnostics)
+        {
+            if (GetUniqueMethodGroupMethod(node, diagnostics) is { } method &&
+                GetMethodGroupOrLambdaDelegateType(method.RefKind, method.ReturnsVoid ? default : method.ReturnTypeWithAnnotations, method.ParameterRefKinds, method.ParameterTypesWithAnnotations, diagnostics) is { } delegateType)
+            {
+                return (delegateType, method);
+            }
+            return default;
+        }
+
+        private MethodSymbol? GetUniqueMethodGroupMethod(BoundMethodGroup node, BindingDiagnosticBag diagnostics)
+        {
+            if (node.Methods.Length > 1)
+            {
+                return null;
+            }
+            var method = node.Methods.SingleOrDefault();
+            if (node.SearchExtensionMethods)
```

Comment:
Should we even look at extension methods if we've found exactly one instance method already?